app/views/assessments/_edit_problems.html.erb

```diff
@@ -1,7 +1,35 @@
-<script type="text/javascript">
-  is_positive_grading = <%= @is_positive_grading %>;
-  has_annotations = <%= @has_annotations %>;
-</script>
+<% content_for :javascripts do %>
+    <script type="text/javascript">
+        is_positive_grading = <%= @is_positive_grading %>;
+        has_annotations = <%= @has_annotations %>;
+
+        document.addEventListener('DOMContentLoaded', function() {
+            var starToggleIcons = document.querySelectorAll('.star-icon');
+
+            starToggleIcons.forEach(function(icon) {
+                icon.addEventListener('click', function() {
+                    toggleStar(icon);
+                });
+            });
+        });
+
+        function toggleStar(icon) {
+            var starred = icon.getAttribute('data-starred');
+            starred = (starred === 'yes') ? 'no' : 'yes';
+            icon.innerHTML = (starred === 'yes') ? 'star' : 'star_border';
+            icon.setAttribute('data-starred', starred);
+            var starred_bool = (starred==='yes') ? 'true' : 'false';
+            // Update the corresponding problem.starred attribute
+            var problemId = icon.getAttribute('data-problem-id');
```

Comment:
Nit: `problemId` is unused, can be removed

Reply:
Done